analyses/create-corplots/1-process-data_VAF.Rmd

```diff
@@ -0,0 +1,571 @@
+# root_dir <- "/Users/chronia/CHOP/GitHub/pbta-tumor-evolution"
+setwd(root_dir)
+
+analysis_dir <- file.path(root_dir, "analyses", "create-corplots")
+
+# File path to input directory
+input_dir <-
+  file.path(analysis_dir, "input")
+
+# Inputs
+input_snv <- file.path(root_dir, "data/snv-consensus-plus-hotspots.maf.tsv.gz")
+
+# The following files were generated from the add-sample-distribution analysis
+input_cohort_pbta <- file.path(input_dir, "pbta.tsv")
+input_genomic_df_filter <- file.path(input_dir, "genomic_df_filter.tsv")
+input_tmb <- file.path(input_dir, "snv-mutation-tmb-coding.tsv")
+
+# File path to results directory
+results_dir <-
+  file.path(analysis_dir, "results")
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+
+# File path to results directory
+plots_dir <-
+  file.path(analysis_dir, "plots")
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
+
+source(paste0(root_dir, "/figures/theme.R"))
+```
+
+## Load data
+```{r load-inputs-please-wait}
+# Inputs
+## Read MAF file
+snv <- data.table::fread(input_snv, data.table = F)
+
+# read pbta cohort
+pbta <- readr::read_tsv(input_cohort_pbta, guess_max = 100000, show_col_types = FALSE)
+
+## Read in genomic_df_filter file
+genomic_df_filter <- readr::read_tsv(input_genomic_df_filter, guess_max = 100000, show_col_types = FALSE)
+
+# Read file with TMB information
+tmb <- readr::read_tsv(input_tmb, guess_max = 100000, show_col_types = FALSE)
+```
+
+
+# Process Data: snv 
+
+We need to process each input file and generate input files.
+We will first change the column name of the biospecimen column to match the column name from the histologies files.
+
+```{r process snv}
+colnames(snv)[which(names(snv) == "Tumor_Sample_Barcode")] <- "Kids_First_Biospecimen_ID"
+```  
+
+```{r calculate VAF, echo=TRUE}
+# using formula for VAF calculation
+snv_vaf <- snv %>% mutate(VAF = t_alt_count / (t_ref_count + t_alt_count))
+```
+
+```{r echo=TRUE}
+# we will subset to variables we will interested to focus on for downstream analysis
+snv_vaf_subset <- subset(snv_vaf, 
+                             select = c(Kids_First_Biospecimen_ID, Hugo_Symbol,Chromosome, Start_Position, Variant_Classification,VAF))
+```
+
+* TODO *
+The following code is part of the module "tumor-clone-inference" and generates the "genomic_df_subset.tsv" file.
+should I remove this part from here (or vice versa) so it's not duplicate in the repo? 
+
+```{r echo=TRUE}
+# Merge genomic_df_filter with pbta_subset to get sample_id information
+# Create match_id2 to count for multiple samples per patient, if any
+genomic_df_filter_merge <- genomic_df_filter %>% 
+  left_join(pbta, by=c('Kids_First_Participant_ID'))
+
+genomic_df_subset <- genomic_df_filter_merge %>%
+  subset(select = c(Kids_First_Participant_ID, Kids_First_Biospecimen_ID, sample_id, composition, match_id, tumor_descriptor, descriptors, cancer_group.x, experimental_strategy.y)) %>%
```

Comment:
the `match_id` in this PBTA file will not work because it is possible to have two different diagnoses over time. How did you come up with this? You can directly use this code: https://github.com/d3b-center/OpenPedCan-analysis/blob/74f73764ce56b3d580d825563e9b0a8709d734fe/analyses/molecular-subtyping-HGG/07-HGG-molecular-subtyping-combine-table.Rmd#L63-L65

We can review this in more detail because this is something you do not want to save in files - it should be created in this code because these sorts of things are always be subject to change with the changing histologies file.